core/os/android/adb/device.go

```diff
@@ -46,13 +46,18 @@ var (
 	devInfoProvidersMutex sync.Mutex
 
 	// cache is a map of device serials to fully resolved bindings.
-	cache      = map[string]*binding{}
+	cache      = map[string]*bindingAndStatus{}
 	cacheMutex sync.Mutex // Guards cache.
 
 	// Registry of all the discovered devices.
 	registry = bind.NewRegistry()
 )
 
+type bindingAndStatus struct {
```

Comment:
Do you need this? Couldn't you just get the status from the binding?

Reply:
Done. Haha, yes, of course, silly me.